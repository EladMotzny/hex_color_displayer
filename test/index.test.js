import React from 'react'
import HexColorDisplay from '../src/index'
import {render} from '@testing-library/react'

// ====================================
// Test Fixtures
// ====================================
const data_example = [
    {
        "start": 0,
        "end": 13,
        "name": "Header",
        "color": "", // Color feature has not yet implemented
        "sublist": []
    },
]
const raw = [54,22,3,6,23,6,1,2,3,4,5,6,7,8,9,10,11,12,13];

// ====================================
// Tests
// ====================================

describe('Test HexColorDisplay sanpshot', () =>{
    it('Matches the snapshot',  () => {
        const component = render(<HexColorDisplay offsets={data_example} bin={raw} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})


