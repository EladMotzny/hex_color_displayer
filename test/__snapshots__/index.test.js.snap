// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HexColorDisplay sanpshot Matches the snapshot 1`] = `
<div>
  <ul>
    <div
      className="row"
    >
      <div
        style={
          {
            "display": "inline-block",
            "width": "4em",
          }
        }
      >
        Address:
      </div>
      <div
        className="hex_row"
      >
        <li
          className="header_item_disabled"
        >
          0
        </li>
        <li
          className="header_item_disabled"
        >
          1
        </li>
        <li
          className="header_item_disabled"
        >
          2
        </li>
        <li
          className="header_item_disabled"
        >
          3
        </li>
        <li
          className="header_item_disabled"
        >
          4
        </li>
        <li
          className="header_item_disabled"
        >
          5
        </li>
        <li
          className="header_item_disabled"
        >
          6
        </li>
        <li
          className="header_item_disabled"
        >
          7
        </li>
        <li
          className="header_item_disabled"
        >
          8
        </li>
        <li
          className="header_item_disabled"
        >
          9
        </li>
        <li
          className="header_item_disabled"
        >
          a
        </li>
        <li
          className="header_item_disabled"
        >
          b
        </li>
        <li
          className="header_item_disabled"
        >
          c
        </li>
        <li
          className="header_item_disabled"
        >
          d
        </li>
        <li
          className="header_item_disabled"
        >
          e
        </li>
        <li
          className="header_item_disabled"
        >
          f
        </li>
      </div>
      <div
        className="heading"
      >
        Dump
      </div>
    </div>
  </ul>
  <ul>
    <div
      className="row"
    >
      <div
        className="heading"
      >
         
        000000:
         
      </div>
      <div
        className="hex_row"
      >
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          36
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          16
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          3
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          6
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          17
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          6
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          1
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          2
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          3
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          4
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          5
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          6
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          7
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          8
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          9
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          a
        </li>
      </div>
      <div
        className="hex_row"
      >
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          6
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="heading"
      >
         
        000010:
         
      </div>
      <div
        className="hex_row"
      >
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          b
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          c
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          d
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
      </div>
      <div
        className="hex_row"
      >
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
        <li
          onMouseOver={[Function]}
          style={
            {
              "backgroundColor": "",
              "color": "",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "width": "1.5em",
            }
          }
        >
          .
        </li>
      </div>
    </div>
  </ul>
</div>
`;
